_tuning-your-cluster/pull-based-ingestion.md

```diff
@@ -0,0 +1,315 @@
+---
+
+# Pull-based ingestion
+**Introduced 3.0**
+{: .label .label-purple }
+
+
+Differently from the [RPC-based ingestion](https://opensearch.org/docs/latest/getting-started/ingest-data/), pull-based ingestion is a way to ingest data into OpenSearch from streaming sources such as Kafka and Kinesis. Pull-based ingestion provides exactly-once ingestion semantics and natively handles backpressure, so it allows the user to process the spike traffic smoothly without worrying about overloading the server.
+
+**Note:**
+Pull-based ingestion is an experimental feature and not fully ready for production yet.
+{: .note}
+
+## Requirements
+
+* Ingestion plugin for respective streaming source needs to be included for using pull-based ingestion. Currently available plugins include **ingestion-kafka** and **ingestion-kinesis**.
+* Pull-based ingestion is supported with segment replication and remote store.
+
+## Create index with pull-based ingestion
+
+To create a new index with pull-based ingestion, you can use the index creation API. For example, if you have a Kafka topic, you can create the index using the following request.
+
+```json
+PUT -H 'Content-Type: application/json' http://localhost:9200/my-index --data ' 
+{
+  "settings": {
+    "ingestion_source": {
+      "type": "kafka",
+      "pointer.init.reset": "earliest",
+      "param": {
+        "topic": "test",
+        "bootstrap_servers": "localhost:49353"
+      }
+    },
+    "index.number_of_shards": 1,
+    "index.number_of_replicas": 1,
+    "index": {
+      "replication.type": "SEGMENT"
+    }
+
+  },
+  "mappings": {
+    "properties": {
+      "name": {
+        "type": "text"
+      },
+      "age": {
+        "type": "integer"
+      }
+    }
+  }
+}'
+```
+{% include copy-curl.html %}
+
+The setting has the following components under ingestion_source:
+* **type**: The pull-based ingestion plugin type, currently `kafka` and `kinesis` are supported.
+* **pointer.init.reset**: Configuration of initial reset to the pointer in the stream, which can be `earliest`, `latest`, `rewind_by_offset`, `rewind_by_timestamp`, `none`.
+* **pointer.init.reset.value**: Only required when the "pointer.init.reset" is set to  `rewind_by_offset` or `rewind_by_timestamp`, this field represents the offset value or timestamp value in milliseconds.
+* **error_strategy**: Indicates the error strategy if a message fails to be processed. Currently supported values include `DROP` and `BLOCK`. By default, failed messages will be dropped. Ingestion can be paused on errors using the "BLOCK" strategy.
+* **max_batch_size**: Defines the max poll size per batch.
+* **poll.timeout**: Defines the poll timeout
+* **num_processor_threads**: Defines the number of writer threads. Default value is 1.
+* **param**: A map of configurations specific to a plugin. For example, Kafka plugin has the topic and bootstrap_servers, also this param can be used to directly configure Kafka consumers.
```

Comment:
It might be worth distinguishing `params` from the other possible properties. The others apply to any ingestion source, whereas `params` is implementation-specific.